### PR TITLE
*: fix weak consistency not work

### LINKED ISSUE #31933: Weak read consistency cannot take effect
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Enable weak read consistency (`set @@tidb_read_consistency = weak`) and execute some queries.
https://github.com/pingcap/tidb/pull/30943 set the isolation under some conditions, however the field will be overwritten by `SetFromSessionVars`.

### 2. What did you expect to see? (Required)
The [isolation field](https://github.com/pingcap/kvproto/blob/805ac3a35f883d87bf1fe70480ef5a7b9e64cd8f/proto/kvrpcpb.proto#L675) of rpc requests should be `RC`.

### 3. What did you see instead (Required)
It's still `SI`

### 4. What is your TiDB version? (Required)

master, release-5.4

### PR DESCRIPTION
**Pull request by you06**
Signed-off-by: you06 <you1474600@gmail.com>

### What problem does this PR solve?

Issue Number: close #31933

Problem Summary:

### What is changed and how it works?

#30943 allows TiDB read by TiKV's RC but with a wrong implementation. This PR merges the RC usage into `SetFromSessionVars` and fix the issue that the weak-consistency read does not work.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix the issue that the weak-consistency read does not work.
```

### CODE REVIEW

File: sessionctx/stmtctx/stmtctx_test.go [hunk 1]
```diff
@@ -96,49 +98,44 @@ func TestWeakConsistencyRead(t *testing.T) {
...
-	lastWeakConsistency := func(tk *testkit.TestKit) bool {
-		return tk.Session().GetSessionVars().StmtCtx.WeakConsistency
-	}
-
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create table t(id int primary key, c int, c1 int, unique index i(c))")
+
+	execAndCheck := func(sql string, rows [][]interface{}, isolationLevel kv.IsoLevel) {
```
Comment (cfzjywxk) [inline sessionctx/stmtctx/stmtctx_test.go:106]: Will this check be executed for each kv request?

**Review by cfzjywxk — APPROVED** [review]

**Review by youjiali1995 — APPROVED** [review]

File: sessionctx/stmtctx/stmtctx_test.go (hunk 1 above)
Comment (you06) [reply sessionctx/stmtctx/stmtctx_test.go:106]: The hook function will be executed in distsql only.

**Review by zyguan — APPROVED** [review]

**Pull request merged**